src/main/java/edu/harvard/iq/dataverse/ingest/plugin/spi/ServiceRegistry.java

```diff
@@ -107,46 +102,47 @@
  * com.mycompany.mypkg.MyService
  * </pre>
  *
- * <p> The service provider classes should be to be lightweight and
- * quick to load.  Implementations of these interfaces should avoid
- * complex dependencies on other classes and on native code. The usual
- * pattern for more complex services is to register a lightweight
- * proxy for the heavyweight service.
+ * <p>
+ * The service provider classes should be to be lightweight and quick to load.
+ * Implementations of these interfaces should avoid complex dependencies on
+ * other classes and on native code. The usual pattern for more complex services
+ * is to register a lightweight proxy for the heavyweight service.
  *
- * <p> An application may customize the contents of a registry as it
- * sees fit, so long as it has the appropriate runtime permission.
+ * <p>
+ * An application may customize the contents of a registry as it sees fit, so
+ * long as it has the appropriate runtime permission.
  *
- * <p> For more details on declaring service providers, and the JAR
- * format in general, see the <a
+ * <p>
+ * For more details on declaring service providers, and the JAR format in
+ * general, see the <a
  * href="../../../../technotes/guides/jar/jar.html">
  * JAR File Specification</a>.
  *
  * @see RegisterableService
  *
  */
-public class ServiceRegistry {
+public class ServiceRegistry <E>{
```

Comment:
@bencomp stolen? 😄 I'm not sure what you mean. We're thinking about using SPI for modularity in Dataverse. See the "Modularity and SPI  Breakout" slides at https://drive.google.com/file/d/0Bzfsw44DUSeZeVF2MmZQTFluaFE/view?usp=sharing and notes at https://docs.google.com/document/d/1JYSkqZ8wnALaeMWRwiRIhhm8NMNLGNFDjmVWUxjd6R0/edit?usp=sharing from last week's 2017 Dataverse Community Meeting.

@oscardssmith did you choose an `E` to mean "element" as opposed to `T` for "type"? I don't know which is better but I found a list of conventions at https://stackoverflow.com/questions/6008241/java-generics-e-and-t-what-is-the-difference/6008274#6008274